src/main/java/com/nvidia/spark/rapids/jni/CastStrings.java

```diff
@@ -152,6 +154,137 @@ public static ColumnVector fromIntegersWithBase(ColumnView cv, int base) {
+   *                                contain time zone.
+   * @param ansiEnabled             is Ansi mode
+   * @return a timestamp column
+   * @throws IllegalArgumentException if any string in cv has invalid format or the time zone is
+   *                                  non-existed/wrong when ansiEnabled is true
+   * @throws CudfException            if time zone is a DST time zone
+   */
+  public static ColumnVector toTimestamp(ColumnView cv, ZoneId defaultTimeZone, boolean ansiEnabled) {
+    if (!GpuTimeZoneDB.isSupportedTimeZone(defaultTimeZone)) {
+      throw new IllegalArgumentException(String.format("Unsupported timezone: %s",
+              defaultTimeZone.toString()));
+    }
+
+    GpuTimeZoneDB singleton = GpuTimeZoneDB.getInstance();
+    if (!singleton.isLoaded()) {
+      GpuTimeZoneDB.cacheDatabase();
+    }
+
+    Integer tzIndex = singleton.getZoneIDMap().get(defaultTimeZone.normalized().toString());
+
+    try (Table transitions = singleton.getTransitions();
+         ColumnVector tzIndices = singleton.getZoneIDVector();
+         ColumnVector tzShortIDs = singleton.getTimeZoneShortIDs()) {
+      return new ColumnVector(toTimestamp(cv.getNativeView(), transitions.getNativeView(),
+              tzIndices.getNativeView(), tzIndex, ansiEnabled, tzShortIDs.getNativeView()));
+    }
+  }
+
+  /**
+   * Trims and parses a timestamp string column with time zone suffix to a
+   * timestamp column.
+   * Do not use the time zones in timestamp strings.
+   *
+   * Supports the following formats:
+   * `[+-]yyyy*`
+   * `[+-]yyyy*-[m]m`
+   * `[+-]yyyy*-[m]m-[d]d`
+   * `[+-]yyyy*-[m]m-[d]d `
+   * `[+-]yyyy*-[m]m-[d]d [h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
+   * `[+-]yyyy*-[m]m-[d]dT[h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
+   * 
+   * Spark supports the following zone id forms:
+   *   - Z - Zulu time zone UTC+0
+   *   - +|-[h]h:[m]m
+   *   - A short id, see
+   * https://docs.oracle.com/javase/8/docs/api/java/time/ZoneId.html#SHORT_IDS
+   *   - An id with one of the prefixes UTC+, UTC-, GMT+, GMT-, UT+ or UT-,
+   *     and a suffix in the formats:
+   *     - +|-h[h]
+   *     - +|-hh[:]mm
+   *     - +|-hh:mm:ss
+   *     - +|-hhmmss
+   *  - Region-based zone IDs in the form `area/city`, such as `Europe/Paris`
+   *
+   * Unlike Spark, Spark-Rapids currently does not support DST time zones.
+   *
+   * Note: Do not support cast special strings(epoch now today yesterday tomorrow) to timestamp.
+   * Spark31x supports cast special strings while Spark320+ do not supports
+   *
+   * Example:
+   * input = [" 2023", "2023-01-01T08:00:00Asia/Shanghai "]
+   * ts = toTimestampWithoutTimeZone(input, allowTimeZone = true,
+   * allowSpecialExpressions = true, ansiEnabled = false)
+   * ts is: ['2023-01-01 00:00:00', '2023-01-01T08:00:00']
+   * 
+   * @param cv                      The input string column to be converted.
+   * @param allowTimeZone           whether allow time zone in the timestamp
```

Comment:
This function name is `toTimestampWithoutTimeZone` , so it will never use the time zones in the strings.
`allowTimeZone`  means whether allow time zone in the timestamp string. If `allowTimeZone` is true, the time zones are ignored; if `allowTimeZone` is false, then this function will throw exception if has any time zone in the strings.

Reply:
Could we make that a little more clear in the comments then?